engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/ZeebeTaskHeadersValidator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.deployment.model.validation;
+
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeHeader;
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeTaskHeaders;
+import io.camunda.zeebe.protocol.Protocol;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.agrona.Strings;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class ZeebeTaskHeadersValidator implements ModelElementValidator<ZeebeTaskHeaders> {
+
+  private static final Set<String> RESTRICTED_HEADERS =
+      Set.of(
+          Protocol.USER_TASK_ASSIGNEE_HEADER_NAME,
+          Protocol.USER_TASK_CANDIDATE_GROUPS_HEADER_NAME,
+          Protocol.USER_TASK_FORM_KEY_HEADER_NAME);
+
+  @Override
+  public Class<ZeebeTaskHeaders> getElementType() {
+    return ZeebeTaskHeaders.class;
+  }
+
+  @Override
+  public void validate(
+      final ZeebeTaskHeaders element, final ValidationResultCollector validationResultCollector) {
+    element.getHeaders().stream()
+        .map(ZeebeHeader::getKey)
+        .filter(Predicate.not(Strings::isEmpty))
+        .filter(RESTRICTED_HEADERS::contains)
+        .map(key -> String.format("Attribute 'key' may not use restricted header '%s'", key))
```

Comment:
:wrench: I find the error message somewhat confusing. It talks about keys and headers and restrictions.
I would prefer the word "reserved" over "restricted". I would also prefer the more explicit "must not use" over "may not use" - where I don't know what that means in English exactly.

And I think the error message could be somewhat more descriptive. E.g.
"Headers must not use reserved key '%s'" or "Header declaration contained value for '%s'. However this key is reserved for internal use by the workflow engine. Consequently, it must not be overwritten"

Or something like that

Reply:
We could also make it a bit more generic and say that anything starting with `io.camunda.zeebe` would be reserved name to be used by Zeebe

Reply:
>We could also make it a bit more generic and say that anything starting with io.camunda.zeebe would be reserved name to be used by Zeebe

👍 I like this. It means we don't have to change any code for this when adding new reserved headers. 

❓ Should I change the tests to be more general or do you think the specific cases make more sense still?

Reply:
I would make them more general